### DISCUSSION TITLE
Regarding pmd:fileName() function

### DISCUSSION THREAD

**Discussion by allurisriakhil** [pmd/pmd#4629] [Q&A] [answered]
Hi @oowekyala, 

As you suggested in the below ticket i updated my script with xpath expression but the below expression is only identifying the files with .cls extension but I also have .xml files in the same directory which it is not flagging. Is there any limitation of this function where it only looks for .cls? Thanks. 

`/*[starts-with(pmd:fileName(), 'SA_')]`

Ticket ref: https://github.com/pmd/pmd/discussions/4601

**Comment by oowekyala** [comment, marked as answer]
You need to set the language of your rule to `xml` so that it executes on XML files. Preferably use the rule class `DomXPathRule` too:
```xml
<rule name="MyXPathRule"
      language="xml"
      message="A message"
      class="net.sourceforge.pmd.lang.xml.rule.DomXPathRule">

      <properties>
        <property name="xpath">
            <value><![CDATA[
               /*[starts-with(pmd:fileName(), 'SA_')] 
            ]]></value>
        </property>
      </properties>
</rule>
```

**Reply by allurisriakhil** [reply]
Hi @oowekyala ,Thanks for the response I modified the rule as per your suggestion but now the rule is not identifying the file names of neither .xml nor .cls files. I am using this rule with Salesforce Code Analyzer which is still on PMD6.

**Reply by allurisriakhil** [reply]
Hi @oowekyala, Please find the full rule below for reference that i am trying to use

```
<?xml version="1.0" encoding="UTF-8"?>
<ruleset name="CustomCodeStyle"
         xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 https://pmd.sourceforge.io/ruleset_2_0_0.xsd">
    <description>Custom ruleset for checking file naming convention starting with SA_</description>
    <rule name="File name should start with SA_ in xml" class="net.sourceforge.pmd.lang.xml.rule.DomXPathRule" language="xml" message="File name should start with SA_ in xml">
        <description>File name should start with SA_</description>
        <priority>3</priority>
        <properties>
          <property name="version" value="2.0"/>
          <property name="xpath">
            <value>
              <![CDATA[ /*[starts-with(pmd:fileName(), 'SA_')]  ]]>
            </value>
          </property>
        </properties>
        <example>
        <![CDATA[
          <file name="SA_MyClass.xml">...</file> <!-- this is correct -->
          <file name="MyClass.xml">...</file> <!-- this is incorrect -->
        ]]>
       </example>
    </rule>
</ruleset>
```

**Reply by allurisriakhil** [reply]
Hi @oowekyala, Thanks i found out the issue the issue is with the sfdx-scanner config. Now i modified it to allow scanning .xml files. Thanks.
